### DISCUSSION TITLE
FixedTopLayout

### DISCUSSION THREAD

**Discussion by lauraocana** [Telefonica/mistica-design#1906] [Proposals]
### Goal

Necesitamos tener disponible el fixed layout en el top de la página según solicitaban los compañeros de Vivo para la funcionalidad de app para no clientes en #1590

### Context

A día de hoy se va a utilizar este layout de forma global para todas las OBs de Novum.

### Impact

Toda la información desarrollada en la discussion #1590

### Discussion Questions

NA

### Do you have a visual proposal?

https://www.figma.com/design/7IbBtzqL4lRFU7xD7roHjt/App-for-no-clients?node-id=1-689&t=b2Tgr9H1J31CxD3s-1

**Comment by HectorSancho** [comment]
¡Hola!, hemos estado dándole vueltas a esta discussion esta mañana en diseño y más tarde con desarrollo. Lo primero sobre lo que hemos hablado era esa idea de integrar esa variante en el actual fixed footer layout, pero hablando con desarrollo hemos llegado a la conclusión de que el comportamiento es demasiado diferente.

- El **bottom fixed layout** que tenemos ahora es un componente que se mantiene siempre igual en modo overlay. Siempre está ahí independientemente de que haya scroll o no.

- Ese **top layout** es más un contenido (un callout en este caso) que se cambia cuando desaparece del viewport al hacer scroll y aparece con una UI diferente. En resumen, según su posición aparece y desaparece cambiando totalmente su aspecto.

Ese cambio del callout por esa capa, podría hacerse con un [intersection observer](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API).

Por ese motivo creemos que esa pieza por ahora debería ser un extended de Novum. Los developers nos han comentado que técnicamente es algo sencillo de implementar al reducirse a una capa blanca con un button group.

También nos han indicado que sería posible facilitar esa solución del cambio de componente desde Apps Core para que pueda ser implementada de forma más sencilla. 

Creo que lo mejor sería que hablen entre los dos equipos de desarrollo.
